common/utils.js

```diff
@@ -28,15 +28,82 @@ export async function buildConstantByNpy(builder, url) {
   if (!dataTypeMap.has(npArray.dataType)) {
     throw new Error(`Data type ${npArray.dataType} is not supported.`);
   }
-  const dimensions = npArray.shape;
+  let dimensions = npArray.shape;
   const type = dataTypeMap.get(npArray.dataType).type;
   const TypedArrayConstructor = dataTypeMap.get(npArray.dataType).array;
-  const typedArray = new TypedArrayConstructor(sizeOfShape(dimensions));
+  let typedArray = new TypedArrayConstructor(sizeOfShape(dimensions));
   const dataView = new DataView(npArray.data.buffer);
   const littleEndian = npArray.byteOrder === '<';
   for (let i = 0; i < sizeOfShape(dimensions); ++i) {
     typedArray[i] = dataView[`get` + type[0].toUpperCase() + type.substr(1)](
         i * TypedArrayConstructor.BYTES_PER_ELEMENT, littleEndian);
   }
+  // TODO(Wanming): This is a workaround to transpose 'ihwo' to 'hwio',
+  // and will be removed once 'ihwo' filterLayout is supported.
+  if (isDepthwise) {
```

Comment:
If it is a workaround for a specific sample, I'd prefer to move it into the sample. You can add a `transpose` op in the graph after create the constant.

Reply:
The `transpose` op in WebNN only supports transpose a `MLOperand`, that's why I have to do this workaround in `buildConstantByNpy`.

Reply:
Good suggestion! It's my misunderstanding the `transpose` op, I thought it could not transpose a `Constant`.